src/server/NetworkingServer.hpp

```diff
@@ -21,36 +21,41 @@ class NetworkingServer {
         NetworkingServer();
         ~NetworkingServer();
 
-        void Start();
-        void collectClientInputs();
-
-        // Setters
-        void setState(std::vector<Group*> active_groups);
+        client_inputs collectClientInputs();
+        void setState(std::vector<std::shared_ptr<Group>> active_groups);
         void incrementTick();
-
-        // Getters
         std::vector<client_direction_update> getClientDirectionUpdates();
         std::vector<int> getClientIds();
-        client_inputs getClientInputs();
     private:
-        // Methods
         void RealtimeServer();
         void ApiServer();
-
         void DeleteClient(sf::TcpSocket* client, sf::SocketSelector selector);
         void Move(sf::Packet command_packet, sf::Uint32 client_id, sf::Uint32 tick);
         void RegisterClient(sf::TcpSocket& client);
 
-        // Variables
+        // Thread safe mClientGroupUpdates operations
```

Comment:
Would be nicer if this was implemented in a thread safe vector class for mClientGroupUpdates
Ditto for mClientMove and mClientSocketsToIds (unordered_maps)

Reply:
maybe we should just use a third party library for that

Reply:
Yeah maybe, I"ll look around. 
Going to leave it for another PR.